polars/polars-io/Cargo.toml

```diff
@@ -24,7 +24,7 @@ decompress = ["flate2/miniz_oxide"]
 decompress-fast = ["flate2/zlib-ng"]
 dtype-categorical = ["polars-core/dtype-categorical"]
 dtype-date = ["polars-core/dtype-date", "polars-time/dtype-date"]
-dtype-datetime = ["polars-core/dtype-datetime", "polars-core/temporal", "polars-time/dtype-datetime"]
+dtype-datetime = ["polars-core/dtype-datetime", "polars-core/temporal", "polars-time/dtype-datetime", "chrono-tz", "chrono"]
```

Comment:
I tried to, instead do
```suggestion
dtype-datetime = ["polars-core/dtype-datetime", "polars-core/temporal", "polars-time/dtype-datetime"]
timezones = ["chrono-tz", "chrono"]
```
and then, in `write_impl.rs`, applying the following:
```diff
diff --git a/polars/polars-io/src/csv/write_impl.rs b/polars/polars-io/src/csv/write_impl.rs
index d2aaab6b5..f855dc97c 100644
--- a/polars/polars-io/src/csv/write_impl.rs
+++ b/polars/polars-io/src/csv/write_impl.rs
@@ -1,9 +1,9 @@
 use std::io::Write;
 
 use arrow::temporal_conversions;
-#[cfg(feature = "dtype-datetime")]
+#[cfg(feature = "timezones")]
 use chrono::TimeZone;
-#[cfg(feature = "dtype-datetime")]
+#[cfg(feature = "timezones")]
 use chrono_tz::Tz;
 use lexical_core::{FormattedSize, ToLexical};
 use memchr::{memchr, memchr2};
@@ -95,6 +95,7 @@ fn write_anyvalue(f: &mut Vec<u8>, value: AnyValue, options: &SerializeOptions)
                 TimeUnit::Milliseconds => temporal_conversions::timestamp_ms_to_datetime(v),
             };
             match tz {
+                #[cfg(feature = "timezones")]
                 Some(tz) => match tz.parse::<Tz>() {
                     Ok(parsed_tz) => match &options.datetime_format {
                         None => write!(f, "{}", PlTzAware::new(ndt, tz)),
```

However, I then get

```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error { kind: Uncategorized, message: "formatter error" }', /home/marcogorelli/polars-dev/polars/polars-io/src/csv/write_impl.rs:132:6
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Traceback (most recent call last):
  File "/home/marcogorelli/polars-dev/py-polars/t.py", line 19, in <module>
    print(df.write_csv())
  File "/home/marcogorelli/polars-dev/py-polars/polars/internals/dataframe/frame.py", line 2237, in write_csv
    self._df.write_csv(
pyo3_runtime.PanicException: called `Result::unwrap()` on an `Err` value: Error { kind: Uncategorized, message: "formatter error" }
```

Looking into it, but am a bit confused

Reply:
How does the panic relate with the dependencies?

Reply:
I don't know, that's what I'm confused about - there's no panic currently, but it appears if I move `chrono-tz` and `chrono` behind the `timezones` feature

Reply:
I can take a look later before merging.

Reply:
OK, I see what happens (kinda): if I put the `Some` arm under the `timezones` feature, then it'll be skipped, and the other arm will be hit
```
            let formatted = match tz {
                #[cfg(feature = "timezones")]
                Some(tz) => match tz.parse::<Tz>() {
                    Ok(parsed_tz) => parsed_tz.from_utc_datetime(&ndt).format(datetime_format),
                    Err(_) => match temporal_conversions::parse_offset(tz) {
                        Ok(parsed_tz) => parsed_tz.from_utc_datetime(&ndt).format(datetime_format),
                        Err(_) => unreachable!(),
                    },
                },
                _ => ndt.format(datetime_format),
            };
```
And then, if `date_format` contains `'%z'`, then `ndt.format(datetime_format)` fails

So, I need to get the `timezones` feature activated when `match tz` is reached with a tz-aware input